modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -245,12 +222,46 @@ public boolean hasSpace() {
 
             histMap.remove(entry.getKey());
 
-            removed.add(cpEntry);
+            rmv.add(cpEntry);
         }
 
         chp.walFilesDeleted(deleted);
 
-        return removed;
+        return rmv;
+    }
+
+    /**
+     * Calculates indexes of WAL segments covered by last checkpoint.
+     *
+     * @return list of indexes or empty list if there are no checkpoints.
+     */
+    private IgniteBiTuple<Long, Long> calculateWalSegmentsCovered() {
```

Comment:
From my point, if we just setting out only first and last values of covered segments this method can by simplified to 4 - 5 lines.
So, whole class can be much shorter.

Reply:
Most of code in the method covers corner cases: there is no checkpoint in history, no WAL segments were covered by the latest checkpoint.
So I doubt the code could be simplified significantly.